### ISSUE TITLE
Command capture bug in linux_procfs save strategy

### ISSUE THREAD

**Issue by mdeguzis** [tmux-plugins/tmux-resurrect#449]
For example, run:
```
vim --cmd "set pythonthreehome=/usr/bin/python3" /tmp/a 
```

Then check ps:
```
 ps -auwwx | grep vim
deguzim  52230  0.3  0.0 164936 12452 pts/0    S+   17:47   0:00 vim --cmd set pythonthreehome=/usr/bin/python3 /tmp/a   
```

They ideally need to use /proc/<PID>/cmdline:
```
$ for i in /proc/5348*/ 
do
    basename "$i"
    < $i/cmdline xargs -0 zsh -c 'printf "\t%q" "$0" "$@"'
    printf '\n'
done
```

Or just one PID:
```
cat /proc/53482/cmdline | xargs -0 zsh -c 'printf "\t%q" "$0" "$@"'
```

Output:
```
53482
        vim      --cmd   set\ pythonthreehome=/apollo/env/envImprovement/python3.8       /tmp/a
```

This is a valid command:
```
vim     --cmd   set\ pythonthreehome=/apollo/env/envImprovement/python3.8       /tmp/a
```

># man printf
>%q     ARGUMENT is printed in a format that can be reused as shell input, escaping non-print‐
>          able characters with the proposed POSIX $'' syntax.

**Comment by bruno-** [comment]
Try using `linux_procfs` strategy for saving commands:

```tmux
set @resurrect-save-command-strategy "linux_procfs"
```

All the available strategies are in `save_command_strategies/` dir in the repo.

I just realized we don't have any docs for this... if you have some time to briefly explain the steps you did to try this, a PR would be welcome.

**Comment by mdeguzis** [comment]
I still get the non-escaped command with that
```
cat ~/.tmux/resurrect/last
vim --cmd set pythonthreehome=/apollo/env/envImprovement/python3.8 /home/deguzim/.tmux.conf
```

conf:
```
  134 set -g @plugin 'tmux-plugins/tpm'
  135 set -g @plugin 'tmux-plugins/tmux-sensible'
  136 set -g @plugin 'tmux-plugins/tmux-resurrect'
  137 set -g @plugin 'tmux-plugins/tmux-continuum'
  138 set @resurrect-save-command-strategy "linux_procfs"
```

**Comment by mdeguzis** [comment]
I saved again and see now it's the command without args
```
pane    work    1       :project-tests      0       :-      0       :/workplace/deguzim/project/src/project 1       vim     :
pane    work    1       :project-tests      0       :-      1       :/workplace/deguzim/project/src/project 0       zsh     :
```

**Comment by mdeguzis** [comment]
Would you accept a PR to add a new save strategy called "linux_proc"? Reading /proc is a pretty sound method. The command below is transforming the null chars and escaping so it ends up working beautifully.
```
cat /proc/53482/cmdline | xargs -0 zsh -c 'printf "\t%q" "$0" "$@"'
```

**Comment by bruno-** [comment]
Hi,

I'm not on Linux so I can't test this easily.

Yes, a PR for this is welcome.

**Comment by bruno-** [comment]
Just please document why `bash -c` with those `printf` interpolations is used. A link to that stackexchange topic in the comments should suffice.

**Comment by mdeguzis** [comment]
submitted https://github.com/tmux-plugins/tmux-resurrect/pull/450

**Comment by bruno-** [comment]
We just merged #450, so I think this one can be closed.
